scripts/bo_benchmarks/demonstrations/branin_campaign_demonstration.py

```diff
@@ -0,0 +1,358 @@
+import matplotlib.pyplot as plt
+import torch
+from gpcheck.models import GPModel, GPConfig
+from gpcheck.metrics import loo_pseudo_likelihood, importance_concentration
+from sklearn.metrics import r2_score
+from scipy.stats import kendalltau
+
+obj1_name = "branin"
+
+
+def branin(x1, x2):
+    y = float(
+        (x2 - 5.1 / (4 * np.pi**2) * x1**2 + 5.0 / np.pi * x1 - 6.0) ** 2
+        + 10 * (1 - 1.0 / (8 * np.pi)) * np.cos(x1)
+        + 10
+    )
+    return y
+
+
+def interval_score(y_true, lower, upper, alpha=0.05):
+    """
+    Calculate interval score for uncertainty quantification evaluation.
+    
+    For 95% confidence intervals (alpha=0.05), the formula is:
+    f_interval = (upper - lower) + (2/alpha) * max(0, lower - y) + (2/alpha) * max(0, y - upper)
+    
+    Args:
+        y_true: True values
+        lower: Lower confidence bounds
+        upper: Upper confidence bounds 
+        alpha: Confidence level (0.05 for 95% CI)
+    
+    Returns:
+        Array of interval scores (lower is better)
+    """
+    width = upper - lower
+    below_penalty = np.maximum(0, lower - y_true)
+    above_penalty = np.maximum(0, y_true - upper)
+    return width + (2 / alpha) * below_penalty + (2 / alpha) * above_penalty
+
+
+def normal_95ci(mean, std):
+    """Convert mean and standard deviation to 95% confidence intervals."""
+    lower = mean - 1.96 * std
+    upper = mean + 1.96 * std
+    return lower, upper
+
+
+def calculate_ax_cross_validation_metrics(ax_client, trial_index):
+    """Calculate metrics using Ax's built-in cross validation with the main client."""
+    try:
+        # Skip if not enough data for cross-validation
+        if trial_index < 3:
+            return None, None
+        
+        # Check if client has a Bayesian model (not just Sobol)
+        if ax_client.generation_strategy.current_step_index == 0:
+            print("  Main client still in Sobol phase, no CV available yet")
+            return None, None
+            
+        # Get the model bridge from the generation strategy 
+        model_bridge = ax_client.generation_strategy.model
+        if model_bridge is None:
+            print("  Main client model bridge is None")
+            return None, None
+        
+        # Run cross validation using Ax's built-in function with the model bridge
+        cv = cross_validate(model=model_bridge, folds=-1)  # Leave-one-out CV
+        fig = interact_cross_validation_plotly(cv)
+        y_act = fig["data"][1].x
+        y_pred = fig["data"][1].y
+        
+        # Calculate R² using Ax's cross validation predictions
+        ax_cv_r2 = r2_score(y_act, y_pred)
+        print(f"  Ax CV R²: {ax_cv_r2:.4f}")
+        
+        # For interval score calculation, we need uncertainty estimates
+        # We'll use the residuals to estimate uncertainty
+        residuals = np.array(y_act) - np.array(y_pred)
+        pred_std = np.std(residuals)  # Simple estimate of prediction uncertainty
```

Comment:
No, you should be getting the uncertainty directly from the plot, similar to how y_act and y_pred are obtained. I'm not sure how exactly to get these values out from the plot, but I'm pretty sure they're there. I'm also not exactly sure what those bounds represent (standard error metric perhaps?) so you might need to check the contents of the Ax 0.5.0 package that you install.